src/main/java/it/gov/pagopa/pu/citizen/service/organization/OrganizationRetrieverServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+
+import it.gov.pagopa.pu.citizen.connector.debtpositions.DebtPositionTypeOrgService;
+import it.gov.pagopa.pu.citizen.dto.generated.OrganizationsWithSpontaneousDTO;
+import it.gov.pagopa.pu.citizen.mapper.OrganizationsWithSpontaneousDTOMapper;
+import it.gov.pagopa.pu.debtpositions.dto.generated.DebtPositionTypeOrgWithActiveSpontaneousCount;
+import it.gov.pagopa.pu.organization.dto.generated.Organization;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@Slf4j
+public class OrganizationRetrieverServiceImpl implements OrganizationRetrieverService{
+
+  private final BrokerOrganizationsRetrieverService brokerOrganizationsRetrieverService;
+  private final DebtPositionTypeOrgService debtPositionTypeOrgService;
+  private final OrganizationsWithSpontaneousDTOMapper organizationsWithSpontaneousDTOMapper;
+
+
+
+  public OrganizationRetrieverServiceImpl(BrokerOrganizationsRetrieverService brokerOrganizationsRetrieverService, DebtPositionTypeOrgService debtPositionTypeOrgService,
+                                          OrganizationsWithSpontaneousDTOMapper organizationsWithSpontaneousDTOMapper
+                                          ) {
+    this.brokerOrganizationsRetrieverService = brokerOrganizationsRetrieverService;
+    this.debtPositionTypeOrgService = debtPositionTypeOrgService;
+    this.organizationsWithSpontaneousDTOMapper = organizationsWithSpontaneousDTOMapper;
+ }
+
+  @Override
+  public List<OrganizationsWithSpontaneousDTO> getOrganizationsWithSpontaneous(Long brokerId, String accessToken) {
+
+    List<Organization> organizations = brokerOrganizationsRetrieverService.getAllOrganizationsByBrokerId(brokerId, null, null, null, accessToken);
+
+    if (organizations.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    List<DebtPositionTypeOrgWithActiveSpontaneousCount> debtPositionTypeOrgWithActiveSpontaneousCount = getDebtPositionTypeOrgWithActiveSpontaneousCounts(accessToken, organizations);
+
+    return organizationsWithSpontaneousDTOMapper.map(getOrganizationsFiltered(organizations, debtPositionTypeOrgWithActiveSpontaneousCount));
+  }
+
+  private List<DebtPositionTypeOrgWithActiveSpontaneousCount> getDebtPositionTypeOrgWithActiveSpontaneousCounts(String accessToken, List<Organization> organizations) {
+    List<Long> organizationsIds = organizations.stream().map(Organization::getOrganizationId).toList();
+    return debtPositionTypeOrgService.getDebtPositionTypeOrgWithActiveSpontaneousCount(organizationsIds, accessToken);
+  }
+
+  private List<Organization> getOrganizationsFiltered( List<Organization> organizationList, List<DebtPositionTypeOrgWithActiveSpontaneousCount> debtPositionTypeOrgWithActiveSpontaneousCounts) {
+
+    List<Long> validIds = debtPositionTypeOrgWithActiveSpontaneousCounts
+      .stream()
+      .map(DebtPositionTypeOrgWithActiveSpontaneousCount::getOrganizationId)
+      .toList();
```

Comment:
for the useCase, a Set would be better
```suggestion
    Set<Long> validIds = debtPositionTypeOrgWithActiveSpontaneousCounts
      .stream()
      .map(DebtPositionTypeOrgWithActiveSpontaneousCount::getOrganizationId)
      .toSet();
```